pkg/multiplexer/multiplexer_test.go

```diff
@@ -164,6 +164,22 @@ func TestFetchTimeout(t *testing.T) {
 	server.Close()
 }
 
+func TestFetchCancelled(t *testing.T) {
+	server := startServer(t)
+
+	r := newRequest()
+	r.WithRequestable(newFakeRequestable("http://localhost:9990?fragment=slow"))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	_, err := r.Do(ctx)
+
+	require.EqualError(t, err, "multiplexer request was canceled: context canceled")
+
+	server.Close()
```

Comment:
```suggestion
	server := startServer(t)
	defer server.Close()

	r := newRequest()
	r.WithRequestable(newFakeRequestable("http://localhost:9990?fragment=slow"))

	ctx, cancel := context.WithCancel(context.Background())
	cancel()

	_, err := r.Do(ctx)

	require.EqualError(t, err, "multiplexer request was canceled: context canceled")
```

I know other tests aren't deferring, but thoughts on starting that here?

Reply:
done in 4fe5e294cd2f1f09e00b72579315bd918c44d66b